### PR TITLE
BAVL-71 adding the necessary client for making calls to the prisoner search API.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
This change will allow our service to make calls to the external prisoner search API for checking the details of a prisoner related to video link bookings.

Note the original intention of this change was to generate the types for the prisoner search API.  However open API seemed to have an issue (bug) with generating it and I don't want to spend too long trying to see if there is a solution at this point in time.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -65,8 +65,8 @@ tasks {
 val configValues = mapOf(
   "dateLibrary" to "java8-localdatetime",
   "serializationLibrary" to "jackson",
-  "useBeanValidation" to "false",
-  "enumPropertyNaming" to "UPPERCASE",
+  "enumPropertyNaming" to "original",
+  "useSpringBoot3" to "true",
```
Comment (matt-crowson-moj) [inline build.gradle.kts:69]: Whilst this makes no obvious impact we should be setting these here during generation of API's for spring boot 3 projects.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      .get()
+      .uri("/prisoner/{prisonerNumber}", prisonerNumber)
+      .retrieve()
+      .bodyToMono(Prisoner::class.java)
+      .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
+      .block()
+}
+
+// TODO add additional fields as and when needed e.g. ACTIVE/NOT ACTIVE in prison
+data class Prisoner(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonersearch/PrisonerSearchClient.kt:20]: As per comment in the description ideally I would have generated this type but open API generator was not playing nicely.

File: src/test/resources/application-test.yml [hunk 3]
```diff
@@ -27,3 +27,4 @@ api:
     url:
       hmpps-auth: http://localhost:8090/auth
       locations-inside-prison: http://localhost:8091
+      prisoner-search: http://localhost:8092
```
Comment (timharrison-moj) [inline src/test/resources/application-test.yml:30]: Are we going with incremental ports for test, so
auth - 8090
locations - 8091
prisoner-search - 8092
xxxx - 8093
yyyy - 8094

Comment (matt-crowson-moj) [reply src/test/resources/application-test.yml:30]: That's what I thought to keep things simple.  Open to other suggestions if you have them.

**Pull request merged**